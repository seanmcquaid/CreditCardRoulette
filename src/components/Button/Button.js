import React from "react";
import styled from "styled-components/native";
import { TouchableHighlight, Text, Alert } from "react-native";
import PropTypes from "prop-types";

const Button = ({title, onPress}) => {
    return (
        <ButtonContainer underlayColor="#C5B358" onPress={onPress} testID="button">
            <ButtonText testID="buttonTitle">{title}</ButtonText>
        </ButtonContainer>
    );
};

const ButtonContainer = styled(TouchableHighlight)`
    background-color: #C5B358;
    padding: 16px 30px;
    border-radius : 3px;
    justify-content : center;
    border : 2px solid #000;
    align-items : center;
    margin : 16px;
`;

const ButtonText = styled(Text)`
    font-family : Merriweather-Regular;
    font-size : 16px;
`;

Button.propTypes = {
    title : PropTypes.string.isRequired,
    onPress : PropTypes.func.isRequired,
};

Button.defaultProps = {
    title : "Put a Title Props Here",
    onPress : () => Alert.alert("Add an on click!")
};

export default Button;