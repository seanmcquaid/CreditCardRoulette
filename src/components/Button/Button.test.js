import React from "react";
import {render, act, fireEvent} from "react-native-testing-library";
import Button from "./Button";

describe("<Button/>", () => {
    it("Matches the snapshot correctly", () => {
        const onPress = jest.fn();

        const props = {
            onPress,
            title : "Mock Button"
        };
        
        const button = render(<Button {...props}/>);

        expect(button).toMatchSnapshot();
    });

    it("OnPress works correctly", () => {
        const onPress = jest.fn();

        const props = {
            onPress,
            title : "Mock Button"
        };

        const {getByTestId} = render(<Button {...props}/>);

        fireEvent.press(getByTestId("button"));

        expect(onPress).toHaveBeenCalledTimes(1);
    });
});