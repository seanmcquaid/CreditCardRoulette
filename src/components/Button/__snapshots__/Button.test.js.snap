// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Matches the snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#C5B358",
        "borderColor": "#000",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 2,
        "justifyContent": "center",
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 16,
        "paddingBottom": 16,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 16,
      },
    ]
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Merriweather-Regular",
          "fontSize": 16,
        },
        null,
      ]
    }
    testID="buttonTitle"
  >
    Mock Button
  </Text>
</View>
`;
